src/Microsoft.Azure.WebJobs.Host/Loggers/FunctionInstanceLogEntry.cs

```diff
@@ -49,12 +51,41 @@ public class FunctionInstanceLogEntry
         /// </summary>
         public string ErrorDetails { get; set; }
 
-        /// <summary>Gets or sets the function's argument values and help strings.</summary>        
+        /// <summary>Gets or sets the function's argument values and help strings.
+        /// If this is null, then the event is before binding. </summary>        
         [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2227:CollectionPropertiesShouldBeReadOnly")]
         public IDictionary<string, string> Arguments { get; set; }
 
         /// <summary>Direct inline capture for output written to the per-function instance TextWriter log. 
         /// For large log outputs, this is truncated</summary>
         public string LogOutput { get; set; }
+
+        /// <summary>
+        /// in-memory Property bag that lives for the lifetime of the item. 
+        /// This can be used to help the caller correlate data at various phases. 
+        /// </summary>
+        [JsonIgnore]
+        public IDictionary<string, object> Properties { get; set; }
+
+        /// <summary>
+        /// An in-memory timer to measure the duration during execution. Use <see cref="Duration"/> for persisted cases.
+        /// </summary>
+        [JsonIgnore]
+        public Stopwatch LiveTimer { get; set; }
+
+        /// <summary>
+        /// Function has just started. This is before arguments are bound. 
+        /// </summary>
+        public bool IsStart => this.Arguments == null;
```

Comment:
Yes. I'll make it more obvious. 

---
In reply to: [127044405](https://github.com/Azure/azure-webjobs-sdk/pull/1218#discussion_r127044405) [](ancestors = 127044405)